### PR TITLE
Fix array_finalize for beams

### PR DESCRIPTION
**Pull request by e-koch**
`minajor` -> `minor` in array_finalize for `Beams`.

I found this trying to do arithmetic with a `Beams` object, which also doesn't work correctly. But I'm not sure if we need it to?

### CODE REVIEW

**Comment by keflavich** [comment]
Oops, that's a bug.

Why not have a `majors * minors * 2 * pi / fwhmfactor` test instead of assuming symmetric beams?

**Comment by e-koch** [comment]
I didn't get that far in the test. Here's an updated version, but it's not working. The original `self` object seems to lose its attributes? I don't understand where array_finalize is getting called when doing an operation and my assumption that the original and new `sr` are both accessible is probably wrong

**Comment by keflavich** [comment]
Right.  I don't think arithmetic operations should work on `Beam`s or `Beams`, it doesn't really make a ton of sense.

**Comment by e-koch** [comment]
Multiplication/division could be made to work if applied on the areas. But I can't think of a case where you would want to that.

We should probably raise some sort of `InvalidBeamOperationError`. Right now it happily returns wrong answers without error.

Addition/subtraction can be treated as convolution/deconvolution like it does in `Beam`

**Comment by keflavich** [comment]
on areas: why not just do `Beams.sr * 5` instead of `(Beams*5).sr`?

And, yes, agreed, raising descriptive errors is a good idea

**Comment by e-koch** [comment]
I've defined a number of operations for `Beams` to match (mostly) the operations in `Beam`.

One weird thing in `Beam`: convolution is multiplication, but deconvolution was subtraction? I think making it division makes more sense as the opposite operation

**Comment by keflavich** [comment]
agreed, division and multiplication make more sense as the de/convolve operations

**Comment by e-koch** [comment]
I've kept the `__sub__` operator in `Beam` in case it's a breaking change. Otherwise, mult/div are used.

I'm happy to merge this. Let me know if you have any more comments.

File: radio_beam/multiple_beams.py [hunk 1]
```diff
@@ -325,3 +337,60 @@ def common_beam(self, includemask=None, method='pts', **kwargs):
...
+    def __truediv__(self, other):
+        # Other must be a single beam. Assume dividing is deconvolving
+        # as set of beams with a given beam
+        if not isinstance(other, Beam):
+            raise InvalidBeamOperationError("Division is defined as a "
+                                            "deconvolution of the set of beams"
+                                            " with a given beam. Must be "
+                                            "divided by a Beam object.")
+
+        return Beams(beams=[beam - other for beam in self])
```
Comment (keflavich) [inline radio_beam/multiple_beams.py]: should this be `beam / other` now?

File: radio_beam/tests/test_beams.py [hunk 2]
```diff
@@ -91,6 +218,12 @@ def test_indexing():
     assert np.all(beams[mask].major.value == majors[mask].value)
 
 
+# def test_beams_mult_convolution():
+
```
Comment (keflavich) [inline radio_beam/tests/test_beams.py]: maybe add a comment about the intent of this one and why it's commented out?  Is this something we'll add later?

File: radio_beam/multiple_beams.py (hunk 1 above)
Comment (e-koch) [reply radio_beam/multiple_beams.py]: Yes, this is wrong

**Comment by e-koch** [comment]
@keflavich - I think we should deprecate subtraction-as-deconvolution. I'll add a warning

File: radio_beam/tests/test_beams.py (hunk 2 above)
Comment (e-koch) [reply radio_beam/tests/test_beams.py]: Oops. This should have been deleted

**Pull request merged**